### DISCUSSION TITLE
Full mass matrix

### DISCUSSION THREAD

**Discussion by Sin-tel** [libigl/libigl#1784] [Ideas]
Are there any plans to include the non-diagonalized mass matrix?
I assume this is what is meant with "MASSMATRIX_TYPE_FULL" in the docs, which is currently not implemented.

**Comment by alecjacobson** [comment]
No plans, but it's simple to implement. See for example:
https://github.com/alecjacobson/gptoolbox/blob/master/mesh/massmatrix.m

happy to accept a pr if you end up implementing it

>

**Reply by Sin-tel** [reply]
Thanks for your reply. Unfortunately I am not comfortable enough with C++ to implement this (I'm just using Python bindings). The lumped mass matrix will do for now.
